public/app/features/explore/state/query.ts

```diff
@@ -702,6 +701,103 @@ export const runQueries = createAsyncThunk<void, RunQueriesOptions>(
+ */
+export const runLoadMoreLogsQueries = createAsyncThunk<void, RunLoadMoreLogsQueriesOptions>(
+  'explore/runQueries',
+  async ({ exploreId, absoluteRange }, { dispatch, getState }) => {
+    dispatch(cancelQueries(exploreId));
+
+    const { datasourceInstance, containerWidth, queryResponse } = getState().explore.panes[exploreId]!;
+    const { defaultCorrelationEditorDatasource, scopedVars, showCorrelationEditorLinks } = await getCorrelationsData(
+      getState(),
+      exploreId
+    );
+    const correlations$ = getCorrelations(exploreId);
+
+    let newQuerySource: Observable<ExplorePanelData>;
+
+    const logQueries = queryResponse.logsResult?.queries || [];
+    const queries = logQueries.map((query: DataQuery) => ({
+      ...query,
+      datasource: query.datasource || datasourceInstance?.getRef(),
+    }));
+
+    if (!hasNonEmptyQuery(queries) || !datasourceInstance) {
+      return;
+    }
+
+    // Some datasource's query builders allow per-query interval limits,
+    // but we're using the datasource interval limit for now
+    const minInterval = datasourceInstance?.interval;
+
+    const queryOptions: QueryOptions = {
+      minInterval,
+      // maxDataPoints is used in:
+      // Loki - used for logs streaming for buffer size, with undefined it falls back to datasource config if it supports that.
+      // Elastic - limits the number of datapoints for the counts query and for logs it has hardcoded limit.
+      // Influx - used to correctly display logs in graph
+      // TODO:unification
+      // maxDataPoints: mode === ExploreMode.Logs && datasourceId === 'loki' ? undefined : containerWidth,
+      maxDataPoints: containerWidth,
+    };
+
+    const timeZone = getTimeZone(getState().user);
+    const range = getTimeRange(
+      timeZone,
+      {
+        from: dateTimeForTimeZone(timeZone, absoluteRange.from),
+        to: dateTimeForTimeZone(timeZone, absoluteRange.to),
+      },
+      getFiscalYearStartMonth(getState().user)
+    );
+    const transaction = buildQueryTransaction(exploreId, queries, queryOptions, range, false, timeZone, scopedVars);
+
+    dispatch(changeLoadingStateAction({ exploreId, loadingState: LoadingState.Loading }));
+
+    newQuerySource = combineLatest([runRequest(datasourceInstance, transaction.request), correlations$]).pipe(
+      mergeMap(([data, correlations]) =>
+        decorateData(
+          // Query splitting, otherwise duplicates results
+          data.state === LoadingState.Done ? mergeDataSeries(queryResponse, data) : data,
+          queryResponse,
+          absoluteRange,
+          undefined,
+          queries,
+          correlations,
+          showCorrelationEditorLinks,
+          defaultCorrelationEditorDatasource
+        )
+      )
+    );
+
+    newQuerySource.subscribe({
+      next(data) {
+        if (data.logsResult !== null && data.state === LoadingState.Done) {
+          reportInteraction('grafana_explore_logs_scrolled', {
```

Comment:
🤔 It feels to me like `reportInteraction` call shouldn't be here but rather called by the component. It's a specific user interaction redux shouldn't be aware of.  Same as `grafana_explore_logs_result_displayed` so maybe it can be tackled separately.

Reply:
103% agree. https://github.com/grafana/grafana/issues/80219